File: smartdevices-app/smartdevicesapp.model/src/main/java/de/vogler_engineering/smartdevicesapp/model/entities/message/Message.java
/**
 * Copyright (c) Vogler Engineering GmbH. All rights reserved.
 * Licensed under the MIT License. See LICENSE.md in the project root for license information.
 */
package de.vogler_engineering.smartdevicesapp.model.entities.message;

import java.util.Date;
import java.util.UUID;

import de.vogler_engineering.smartdevicesapp.model.entities.job.Priority;
import lombok.Data;

/**
 * Created by vh on 14.02.2018.
 */

@Data
public class Message {

    protected UUID id;
    protected UUID referenceId;
    protected Priority priority;

    private String name;

    protected Date createdAt;
    protected Date assignedAt;

    protected String createdBy;
    protected String assignedTo;

    protected String type;
}

File: smartdevices-gateway/SmartDevicesGateway.Model/Messages/Message.cs
//
// Copyright (c) Vogler Engineering GmbH. All rights reserved.
// Licensed under the MIT License. See LICENSE.md in the project root for license information.
//
using System;
using SmartDevicesGateway.Model.Jobs.Enums;
using SmartDevicesGateway.Model.Messages.Base;
using SmartDevicesGateway.Model.Messages.Enums;

namespace SmartDevicesGateway.Model.Messages
{
    public class Message : IMessage
    {
        public Guid Id { get; set; }
        public Guid ReferenceId { get; set; }
        public MessagePriority Priority { get; set; }

        public string Name { get; set; }

        public string CreatedBy { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
        public string AssignedTo { get; set; }
        public DateTimeOffset AssignedAt { get; set; }

        public string Type { get; set; }
    }
}